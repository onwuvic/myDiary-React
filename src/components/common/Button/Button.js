import React from 'react';
import PropTypes from 'prop-types';


const Button = ({
  type, children, styles
}) => (
  <button type={type} className={styles}>{children}</button>
);

Button.propTypes = {
  type: PropTypes.string,
  children: PropTypes.node,
  styles: PropTypes.string
};

export { Button };
