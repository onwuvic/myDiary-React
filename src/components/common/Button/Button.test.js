import React from 'react';
import { mount } from 'enzyme';

import Button from './Button';

let component;

beforeEach(() => {
  component = mount(<Button
    type="submit"
    styles="btn btn-info">Save</Button>);
});

afterEach(() => {
  component.unmount();
});

it('should have a button', () => {
  expect(component.find('button').length).toEqual(1);
});
