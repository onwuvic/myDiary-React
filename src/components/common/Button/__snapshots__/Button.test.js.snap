// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> should render without throwing an error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <UNDEFINED />
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": undefined,
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": undefined,
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`should render without throwing an error 1`] = `undefined`;
